### PR TITLE
Added Strategy, Locale Provider, ICU Locale, ICU Rules, and OID options while creating a database. #6383

### PR DESCRIPTION
**Pull request by akshay-joshi**

### CODE REVIEW

File: web/pgadmin/browser/server_groups/servers/databases/static/js/database.ui.js [hunk 1]
```diff
@@ -125,17 +130,90 @@ export default class DatabaseSchema extends BaseUISchema {
...
+        },
       },{
         id: 'datctype', label: gettext('Character type'),
         editable: false, type: 'select', group: gettext('Definition'),
         readonly: function(state) {return !obj.isNew(state); },
         options: this.fieldOptions.datctype,
+        deps: ['datlocaleprovider'],
+        depChange: (state)=>{
+          if (state.datlocaleprovider !== 'libc')
+            state.datctype = '';
```
Comment (adityatoshniwal) [inline web/pgadmin/browser/server_groups/servers/databases/static/js/database.ui.js:180]: same as above

File: web/pgadmin/browser/server_groups/servers/databases/static/js/database.ui.js [hunk 2]
```diff
@@ -125,17 +130,90 @@ export default class DatabaseSchema extends BaseUISchema {
...
+        },
       },{
+        id: 'daticulocale', label: gettext('ICU Locale'),
+        editable: false, type: 'select', group: gettext('Definition'),
+        readonly: function(state) {return !obj.isNew(state); },
+        options: this.fieldOptions.daticulocale,
+        deps: ['datlocaleprovider'],
+        depChange: (state)=>{
+          if (state.datlocaleprovider !== 'icu')
+            state.daticulocale = '';
```
Comment (adityatoshniwal) [inline web/pgadmin/browser/server_groups/servers/databases/static/js/database.ui.js:193]: same as above and other places

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]

File: docs/en_US/release_notes_7_7.rst [hunk 3]
```diff
@@ -21,10 +21,12 @@ New features
 ************
 
   | `Issue #4805 <https://github.com/pgadmin-org/pgadmin4/issues/4805>`_ -  Added all the new options of the 'WITH' clause in the subscription dialog.
+  | `Issue #6383 <https://github.com/pgadmin-org/pgadmin4/issues/6383>`_ -  Added Strategy, Locale Provider, ICU Locale, ICU Rules, and OID options while creating a database.
```
[the remark this answers is not in the document]
Comment (akshay-joshi) [reply docs/en_US/release_notes_7_7.rst:24]: It's standard that Grammarly suggested

File: docs/en_US/database_dialog.rst [hunk 4]
```diff
@@ -38,11 +41,22 @@ Use the *Definition* tab to set properties for the database:
 * Select a tablespace from the drop-down listbox in the *Tablespace* field. The
   selected tablespace will be the default tablespace used to contain database
   objects.
+* Select the strategy from the drop-down listbox in the *Strategy* field while
+  creating a new database. This option is available from v15 onwards.
```
[the remark this answers is not in the document]
Comment (akshay-joshi) [reply docs/en_US/database_dialog.rst:45]: This option is only available in create mode.

File: docs/en_US/release_notes_7_7.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (adityatoshniwal) [reply docs/en_US/release_notes_7_7.rst:24]: OK

File: docs/en_US/database_dialog.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (adityatoshniwal) [reply docs/en_US/database_dialog.rst:45]: OK

**Pull request merged**